server/src/main/java/org/opensearch/gateway/PrimaryShardBatchAllocator.java

```diff
@@ -121,6 +122,27 @@ public void allocateUnassignedBatch(List<ShardRouting> shardRoutings, RoutingAll
         logger.trace("Finished shard allocation execution for unassigned primary shards: {}", shardRoutings.size());
     }
 
+    protected void allocateUnassignedBatchOnTimeout(List<ShardRouting> shardRoutings, RoutingAllocation allocation) {
+        Set<ShardRouting> batchShardRoutingSet = new HashSet<>(shardRoutings);
+        RoutingNodes.UnassignedShards.UnassignedIterator iterator = allocation.routingNodes().unassigned().iterator();
+        while (iterator.hasNext()) {
+            ShardRouting unassignedShard = iterator.next();
+            AllocateUnassignedDecision allocationDecision;
+            if (unassignedShard.primary() && batchShardRoutingSet.contains(unassignedShard)) {
+                allocationDecision = new AllocateUnassignedDecision(
+                    UnassignedInfo.AllocationStatus.DECIDERS_THROTTLED,
```

Comment:
This throttling will get reset in next iteration automatically?

Also why is there a need to mark the shards in remaining batches throttled? Just leave them as is and will be picked up in next iteration

Reply:
We mark it throttled because it is an existing shard, and we want to prevent the `BalancedShardsAllocator` from attempting to allocate it during the current allocation round. By marking it throttled, we ensure that this unassigned shard is removed and ignored from consideration in current round. However, it will be added back whenever next reroute is triggered. See https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/routing/allocation/ExistingShardsAllocator.java#L175